integrationtests/controller/bundle/bundle_targets_test.go

```diff
@@ -301,14 +306,98 @@ var _ = Describe("Bundle targets", Ordered, func() {
+		BeforeEach(func() {
+			bundleName = "skip"
+			expectedNumberOfBundleDeployments = 0
+			// simulate targets in fleet.yaml which are used for customization
+			targets = []v1alpha1.BundleTarget{
+				{
+					ClusterGroup: "all",
+					DoNotDeploy:  true,
+				},
+			}
+			// simulate targets in GitRepo. All targets in GitRepo are also added to targetRestrictions, which acts as a white list
+			targetsInGitRepo := []v1alpha1.BundleTarget{
+				{
+					ClusterGroup: "all",
+				},
+			}
+			targetRestrictions = make([]v1alpha1.BundleTarget, len(targetsInGitRepo))
+			copy(targetRestrictions, targetsInGitRepo)
+			targets = append(targets, targetsInGitRepo...)
+		})
+
+		It("no BundleDeployments are created", func() {
+			waitForBundleToBeReady(bundleName)
+			_ = verifyBundlesDeploymentsAreCreated(expectedNumberOfBundleDeployments, bdLabels, bundleName)
+		})
+	})
+
+	When("GitRepo has a target that matches clusterGroup one, and a targetCustomization that matches all clusters has DoNotDeploy set to true", func() {
+		BeforeEach(func() {
+			bundleName = "skipone"
+			expectedNumberOfBundleDeployments = 0
+			// simulate targets in fleet.yaml which are used for customization
+			targets = []v1alpha1.BundleTarget{
+				{
+					ClusterGroup: "all",
+					DoNotDeploy:  true,
+				},
+			}
+			// simulate targets in GitRepo. All targets in GitRepo are also added to targetRestrictions, which acts as a white list
+			targetsInGitRepo := []v1alpha1.BundleTarget{
+				{
+					ClusterGroup: "one",
+				},
+			}
+			targetRestrictions = make([]v1alpha1.BundleTarget, len(targetsInGitRepo))
+			copy(targetRestrictions, targetsInGitRepo)
+			targets = append(targets, targetsInGitRepo...)
+		})
+
+		It("no BundleDeployments are created", func() {
+			waitForBundleToBeReady(bundleName)
+			_ = verifyBundlesDeploymentsAreCreated(expectedNumberOfBundleDeployments, bdLabels, bundleName)
+		})
+	})
+
+	When("GitRepo has a target that matches clusterGroup one, and a targetCustomization that matches clusterGroup two has DoNotDeploy set to true", func() {
+		BeforeEach(func() {
+			bundleName = "dontskip"
+			expectedNumberOfBundleDeployments = 1
+			// simulate targets in fleet.yaml which are used for customization
+			targets = []v1alpha1.BundleTarget{
+				{
+					ClusterGroup: "two",
+					DoNotDeploy:  true,
+				},
+			}
+			// simulate targets in GitRepo. All targets in GitRepo are also added to targetRestrictions, which acts as a white list
+			targetsInGitRepo := []v1alpha1.BundleTarget{
+				{
+					ClusterGroup: "one",
+				},
+			}
+			targetRestrictions = make([]v1alpha1.BundleTarget, len(targetsInGitRepo))
+			copy(targetRestrictions, targetsInGitRepo)
+			targets = append(targets, targetsInGitRepo...)
+		})
+
+		It("one BundleDeployment is created", func() {
+			_ = verifyBundlesDeploymentsAreCreated(expectedNumberOfBundleDeployments, bdLabels, bundleName)
```

Comment:
Why do we not wait for the bundle to be ready here, as we do in the two test cases above this one?
Also, would we want to wait for the bundle to be ready in already existing test cases further up in this file?

Reply:
In this case it will never be ready because we are just testing this controller, so the BundleDeployment won't be ready as other things need to happen outside of the controller. The previous test case will be ready because it has 0 BundleDeployments